onset_maproom/layout.py

```diff
@@ -21,20 +27,73 @@
 IRI_GRAY = "rgb(113,112,116)"
 LIGHT_GRAY = "#eeeeee"
 
+DBPOOL = pingrid.init_dbpool("dbpool", CONFIG)
+
+
+def adm_borders(shapes):
+    dbpool = DBPOOL
+    with dbpool.take() as cm:
+        conn = cm.resource
+        with conn:  # transaction
+            s = sql.Composed(
+                [
+                    sql.SQL("with g as ("),
+                    sql.SQL(shapes),
+                    sql.SQL(
+                        """
+                        )
+                        select
+                            g.label, g.key, g.the_geom
+                        from g
+                        """
+                    ),
+                ]
+            )
+            df = pd.read_sql(
+                s,
+                conn,
+            )
+    df["the_geom"] = df["the_geom"].apply(lambda x: wkb.loads(x.tobytes()))
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    return {"features": shapes}
+
+
+def make_adm_overlay(adm_list, adm_lev, adm_weight):
+    border_id = f'borders_adm{adm_lev}'
+    adm_check = False
+    if adm_lev == 1:
+        adm_check = True
+    return dlf.Overlay(
+        dlf.GeoJSON(
+            id=border_id,
+            data=adm_borders(adm_list[adm_lev-1]["sql"]),
+            options={
+                "fill": False,
+                "color": adm_list[adm_lev-1]["color"],
+                "weight": adm_weight,
+            },
+        ),
+        name=adm_list[adm_lev-1]["name"],
+        checked=adm_check,
+    )
+
 
 def app_layout():
 
     # Initialization
     rr_mrg = calc.read_zarr_data(RR_MRG_ZARR)
     center_of_the_map = [((rr_mrg.Y[0]+rr_mrg.Y[-1])/2).values, ((rr_mrg.X[0]+rr_mrg.X[-1])/2).values]
     lat_res = np.around((rr_mrg.Y[1]-rr_mrg.Y[0]).values, decimals=10)
-    lat_min = str(np.around((rr_mrg.Y[0]-lat_res/2).values, decimals=10))
-    lat_max = str(np.around((rr_mrg.Y[-1]+lat_res/2).values, decimals=10))
+    lat_min = np.around((rr_mrg.Y[0]-lat_res/2).values, decimals=10)
+    lat_max = np.around((rr_mrg.Y[-1]+lat_res/2).values, decimals=10)
     lon_res = np.around((rr_mrg.X[1]-rr_mrg.X[0]).values, decimals=10)
-    lon_min = str(np.around((rr_mrg.X[0]-lon_res/2).values, decimals=10))
-    lon_max = str(np.around((rr_mrg.X[-1]+lon_res/2).values, decimals=10))
-    lat_label = lat_min+" to "+lat_max+" by "+str(lat_res)+"˚"
-    lon_label = lon_min+" to "+lon_max+" by "+str(lon_res)+"˚"
+    lon_min = np.around((rr_mrg.X[0]-lon_res/2).values, decimals=10)
+    lon_max = np.around((rr_mrg.X[-1]+lon_res/2).values, decimals=10)
+    lat_label = str(lat_min)+" to "+str(lat_max)+" by "+str(lat_res)+"˚"
```

Comment:
These labels become too long to properly fit inside the control for selecting the lat/lng - is it possible to round them to a shorter number so that all ten decimal points are not displayed for the control description?

Reply:
yeah... I am not sure what to do about that... The resolution is 0.0375, so getting the edges means to add half the resolution (0.01875) to a the first and last centers that themselves can be already fairly long... (the Ghana case is an extreme case where the nc files were made on not rounded X/Y grid values...).

Maybe should get away from those labels and convey the info about valid lat/lon entries otherwise... Or just rely on the Alerts that are in place.

Reply:
If we need to display the values with accuracy and thus all of the decimal places I would say moving them to the description of the maproom or maybe displaying them when hovering over the input box title?

Reply:
I made this an issue as this is not related to this PR.